stl/inc/random

```diff
@@ -2932,15 +2932,32 @@ private:
             _Ty _Vx1;
             _Ty _Vx2;
             _Ty _Sx;
-            for (;;) { // reject bad values
+            for (;;) { // reject bad values to avoid generating NaN/Inf on the next calculations
                 _Vx1 = 2 * _NRAND(_Eng, _Ty) - 1;
                 _Vx2 = 2 * _NRAND(_Eng, _Ty) - 1;
                 _Sx  = _Vx1 * _Vx1 + _Vx2 * _Vx2;
-                if (_Sx < 1) {
+                if (_Sx < _Ty{1} && _Sx != _Ty{0}) {
+                    // good values!
                     break;
                 }
             }
-            const auto _Fx = static_cast<_Ty>(_CSTD sqrt(-2.0 * _CSTD log(_Sx) / _Sx));
+
+            _Ty _LogSx;
+            if (_Sx > 1.0e-4) {
+                _LogSx = _STD log(_Sx);
+            } else {
+                // Bad _Sx value! It will cause overflow on log calculation

```

Comment:
This is a little misleading, because it must be much smaller to actually overflow. 
```suggestion
                // Bad _Sx value! Very small values will overflow log(_Sx)/_Sx

```